src/rand.rs

```diff
@@ -276,6 +281,26 @@ mod darwin {
     }
 }
 
+#[cfg(target_os = "fuchsia")]
+mod fuchsia {
+    extern crate fuchsia_zircon;
+
+    use error;
+
+    pub fn fill(dest: &mut [u8]) -> Result<(), error::Unspecified> {
+        for s in dest.chunks_mut(fuchsia_zircon::sys::ZX_CPRNG_DRAW_MAX_LEN) {
```

Comment:
We can't yet remove the outer loop. The current syscall guarantees on fuchsia for CPRNG_DRAW don't guarantee a full read, because you might be kicked out of the kernel space.

Reply:
We have an active issue addressing this, and it will be changed.

Reply:
> We can't yet remove the outer loop. The current syscall guarantees on fuchsia for CPRNG_DRAW don't guarantee a full read, because you might be kicked out of the kernel space.

Isn't that handled by the *inner* loop? I agree we need to keep the inner loop, but the outer loop seems unnecessary. In particular, consider this case:

* The buffer to fill is 4/3 * ZX_CPRNG_DRAW_MAX_LEN bytes long.
* Each call to ` fuchsia_zircon::cprng_draw` gives us 2/3 of ZX_CPRNG_DRAW_MAX_LEN bytes.

In such a scenerio, without the outer loop, 2 calls to ` fuchsia_zircon::cprng_draw` are needed. But, with the outer loop, 3 calls to `fuchsia_zircon::cprng_draw` are needed.